src/dbsync/tests/interface/dbsync_test.cpp

```diff
@@ -0,0 +1,128 @@
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT, PRIMARY KEY (`pid`)) WITHOUT ROWID;"};
+   
+    const auto handle { dbsync_initialize(HostType::AGENT, DbEngineType::SQLITE3, DATABASE_TEMP, sql) };
+    ASSERT_NE(nullptr, handle);
+
+    EXPECT_NE(0, dbsync_insert_data(handle, nullptr));
+
+    EXPECT_NO_THROW(dbsync_teardown());
+}
+
+TEST_F(DBSyncTest, InsertDataInvalidHandle) 
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT, PRIMARY KEY (`pid`)) WITHOUT ROWID;"};
+    const auto insert_sql{ R"({"table":"processes","data":[{"pid":4,"name":"System"}]})"};
+    
+    const auto handle { dbsync_initialize(HostType::AGENT, DbEngineType::SQLITE3, DATABASE_TEMP, sql) };
+    ASSERT_NE(nullptr, handle);
+
+    const std::unique_ptr<cJSON, smartDeleterJson> json_insert{ cJSON_Parse(insert_sql) };
+
+    EXPECT_NE(0, dbsync_insert_data(reinterpret_cast<void *>(0xffffffff), json_insert.get()));
+
+    EXPECT_NO_THROW(dbsync_teardown());
+}
+
+TEST_F(DBSyncTest, UpdateData) 
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT, PRIMARY KEY (`pid`)) WITHOUT ROWID;"};
+    const auto insert_sql{ R"({"table":"processes","data":[{"pid":4,"name":"System"}]})"};
+    
+    const auto handle { dbsync_initialize(HostType::AGENT, DbEngineType::SQLITE3, DATABASE_TEMP, sql) };
+    ASSERT_NE(nullptr, handle);
+
+    const std::unique_ptr<cJSON, smartDeleterJson> json_insert{ cJSON_Parse(insert_sql) };
+
+    cJSON * json_response { nullptr };
+    
+    EXPECT_EQ(0, dbsync_update_with_snapshot(handle, json_insert.get(), &json_response));
+    EXPECT_NE(nullptr, json_response);
+    EXPECT_NO_THROW(dbsync_free_result(&json_response));
+
+    EXPECT_NO_THROW(dbsync_teardown());
+}
+
+TEST_F(DBSyncTest, FreeNullptrResult) 
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT, PRIMARY KEY (`pid`)) WITHOUT ROWID;"};
+    
+    const auto handle { dbsync_initialize(HostType::AGENT, DbEngineType::SQLITE3, DATABASE_TEMP, sql) };
+    ASSERT_NE(nullptr, handle);
+
+    cJSON* json_response { nullptr };
+
+    EXPECT_NO_THROW(dbsync_free_result(&json_response));
+
+    EXPECT_NO_THROW(dbsync_teardown());
+}
+
+TEST_F(DBSyncTest, UpdateDataWithLessFields) 
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT,`path` TEXT, PRIMARY KEY (`pid`)) WITHOUT ROWID;"};
+    const auto insert_sql{ R"({"table":"processes","data":[{"pid":4,"name":"System"}]})"};
+    
+    const auto handle { dbsync_initialize(HostType::AGENT, DbEngineType::SQLITE3, DATABASE_TEMP, sql) };
+    ASSERT_NE(nullptr, handle);
+
+    const std::unique_ptr<cJSON, smartDeleterJson> json_insert{ cJSON_Parse(insert_sql) };
+
+    cJSON * json_response { nullptr };
+    
+    EXPECT_EQ(0, dbsync_update_with_snapshot(handle, json_insert.get(), &json_response));
+    EXPECT_NE(nullptr, json_response);
+    EXPECT_NO_THROW(dbsync_free_result(&json_response));
+
+    EXPECT_NO_THROW(dbsync_teardown());
```

Comment:
I think this EXPECT could be moved to the teardown method from gtest. All of these are test fixture so that teardown will be executed at the end of each test.